### ISSUE TITLE
MultiHeadMLPLearner.update never validates targets shape: a length-1 target silently trains every head as active

### ISSUE THREAD

**Issue by rama0x1** [SlopDotCash/asi#452]
`MultiHeadMLPLearner.update(state, observation, targets)` documents `targets: Per-head targets, shape (n_heads,). NaN = inactive head.` but never checks the shape before use. When `targets` is shorter than `n_heads` (a scalar or length-1 array is the natural accident — e.g. a caller migrating from a single-head learner, or building the array with the wrong axis), JAX's static integer indexing on the per-head Python loop (`for i in range(n_heads): ... safe_targets[i] ...`) silently **clamps** the out-of-range index to the last valid one instead of raising. The result: every head is treated as active and trained on the same clamped value, and every head reports a finite (non-NaN) `squared_error`/`error`, exactly as if the caller had supplied a full `(n_heads,)` array with every head intentionally active.

This is the same defect class already fixed in `UPGDLearner.update` by #402 (`core/upgd.py`), just left open in the shared-trunk `MultiHeadMLPLearner` that `HordeLearner`, `CBPMultiHeadMLPLearner`, and `SARSAAgent` all build on.

## Reproduction

```python
import jax.numpy as jnp
import jax.random as jr
from alberta_framework.core.multi_head_learner import MultiHeadMLPLearner

learner = MultiHeadMLPLearner(n_heads=4, hidden_sizes=(8,), sparsity=0.0, step_size=0.01)
state = learner.init(feature_dim=3, key=jr.key(0))
obs = jnp.array([1.0, -0.5, 0.25])

# Caller intends: only head 0 active, target 2.0; heads 1-3 inactive (NaN).
# Accidentally passes a length-1 array instead of jnp.array([2.0, nan, nan, nan]).
result = learner.update(state, obs, jnp.array([2.0]))
print(result.errors)          # [2.78 1.31 1.85 1.33]  <- all four heads "active"
print(result.per_head_metrics[:, 0])  # [7.74 1.72 3.44 1.76] <- 3 phantom squared errors
```

Expected: `ValueError` (or at minimum NaN for heads 1-3, matching the documented "NaN = inactive head" contract). Actual: all four heads silently receive the same clamped value and report a finite squared error, corrupting any tracking-error / loss metric computed downstream (this learner backs `HordeLearner`, `SARSAAgent`, and `CBPMultiHeadMLPLearner`).

Other invalid shapes (`()`, `(4, 1)`, `(5,)`) already raise a (fairly opaque) `IndexError`/`TypeError`/`ValueError` from deeper in the trace, so this is specifically about the length-1-or-shorter case, which is the one that succeeds silently.

## Fix

Add the same static (nil-jit-cost) shape check used by `UPGDLearner.update`:

```python
targets = jnp.asarray(targets, dtype=jnp.float32)
if targets.shape != (n_heads,):
    raise ValueError(f"targets must have shape ({n_heads},), got {targets.shape}")
```

## Falsifying test

```
.venv/bin/python -m pytest tests/test_multi_head_learner.py::TestMultiHeadUpdateValidation -q -o addopts=""
```

fails on `main` (no error raised, or a different exception than the intended one) and should pass once the shape check lands.
